webviz_config/webviz_plugin_subclasses/_callback_typecheck.py

```diff
@@ -0,0 +1,64 @@
+from typing import Any, Callable, get_origin, TypeVar
+import inspect
+from enum import Enum
+
+T = TypeVar("T")
+
+
+class ConversionError(Exception):
+    pass
+
+
+def convert(arg: Any, convert_to: T) -> T:
+    # pylint: disable=too-many-return-statements
+    additional_error_message: str = ""
+    try:
+        if inspect.isclass(convert_to) and issubclass(convert_to, Enum):  # type: ignore[arg-type]
+            return convert_to(arg)  # type: ignore[operator]
+        if convert_to is int:
+            return int(arg)  # type: ignore[return-value]
+        if convert_to is float:
+            return float(arg)  # type: ignore[return-value]
+        if convert_to is str:
+            return str(arg)  # type: ignore[return-value]
+        if convert_to is list and isinstance(arg, list):
+            return arg  # type: ignore[return-value]
+        if get_origin(convert_to) is list and isinstance(arg, list):
+            if "__args__" in dir(convert_to):
+                return [convert(a, convert_to.__args__[0]) for a in arg]  # type: ignore[attr-defined,return-value]
+        if convert_to is dict and isinstance(arg, dict):
+            return arg  # type: ignore[return-value]
+        if get_origin(convert_to) is dict and isinstance(arg, dict):
+            if "__args__" in dir(convert_to) and len(convert_to.__args__) == 2:  # type: ignore[attr-defined]
+                return {  # type: ignore[return-value]
+                    convert(key, convert_to.__args__[0]): convert(  # type: ignore[attr-defined]
+                        value, convert_to.__args__[1]  # type: ignore[attr-defined]
+                    )
+                    for key, value in arg.items()
+                }
```

Comment:
It might be that this can be outsourced to `pydantic` at some point. Basically it is `json`-types to a Python type? If so tagging #357 and #199 to couple them to this code snippet.